src/coins.cpp

```diff
@@ -782,6 +829,9 @@ bool CCoinsViewCache::RestoreImmatureBalances(int blockHeight, const CBlockUndo&
 
             cacheSidechains.at(scId).flag = CSidechainsCacheEntry::Flags::DIRTY;
         }
+
+        cacheSidechains.at(scId).scInfo.lastCertificateHash = lastCertHash;
```

Comment:
for sake of uniformity this should be removed from here and added to 827 above, together with the setting of targetScInfo.lastEpochReferencedByCertificate, that would also spare a useless DIRTY set